Tests for review_related model and schema

Jeshan7 opened the pull request:
## Why?
 - Test for product_review and review(schema) was not present.
 - Factory for review and product review was not present.
 - Refactored the product_review and review (model tests).

## This change addresses the need by:
  - Test for product_review and review are added.(schema)
 - Factory for review and product_review is now present.
 - Refactored the product_review and review(model tests).
 - Made a change in review.ex (schema). user_id was needed to be included
   but it was not included in update_params.
   (Added user_id attribute to @update_params)

[delivers #163622129]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163622129]

Prior to this change,
 - Test for schema of product_review and review were not present.
 - Factory for review and product review was not present.
 - Refactored the product_review and review (model tests).

This change
 - Test for product_review and review are added.(schema)
 - Factory for review and product_review is now present.
 - Refactored the product_review and review(model tests).
 - Made a change in review.ex (schema). user_id was needed to be included
   but it was not included in update_params.
   (Added user_id attribute to @update_params)

arjun289 commented on apps/snitch_core/lib/core/data/schema/review/review.ex:
```diff
@@ -35,7 +35,7 @@ defmodule Snitch.Data.Schema.Review do
 
   @create_params @required_params ++ @optional_params
-  @update_params ~w(description)a ++ @optional_params
+  @update_params ~w(description user_id)a ++ @optional_params
```
please remove `user_id` user_id should not be updated for a review.

Jeshan7 replied:
@arjun289 sir But if we don't include user_id in updated params then how will we able to check foreign key constraint property condition.

arjun289 replied:
It needs to be checked only during `create` action, no need to check it during the update.

arjun289 commented on apps/snitch_core/test/data/model/product_review_test.exs:
```diff
@@ -42,6 +42,7 @@ defmodule Snitch.Data.Model.ProductReviewTest do
     %{rating_options: rating_options} = context
 
     [rating_option_1 | [rating_option_2 | _]] = rating_options
+    IO.inspect(rating_option_1.value)
```
remove `IO.inspect`

arjun289 commented on apps/snitch_core/test/data/model/product_review_test.exs:
```diff
@@ -55,6 +56,20 @@ defmodule Snitch.Data.Model.ProductReviewTest do
+  describe "fetch_rating_option/1" do
+    test "return a valid rating_option", context do
+      %{rating_options: rating_options} = context
+      [rating_option_1 | [rating_option_2 | _]] = rating_options
```
this can be written simply as [rating_option_1, rating_option_2] since you are inserting only two rating options and you are controlling it using tags.

arjun289 commented on apps/snitch_core/test/data/model/review_test.exs:
```diff
@@ -5,58 +5,52 @@ defmodule Snitch.Data.Model.ReviewTest do
+      [review, review_2] = reviews
+      rpv = review_2.rating_option_vote
+      rating_option_vote = Map.from_struct(rpv)
+
```
You are making use of rating option vote from another review. Instead, just insert a rating_option_vote and use that.

Jeshan7 replied:
@arjun289 If we try to insert a rating option separately then we will get an error regarding the uniqueness of rating option as we are trying to insert same rating option twice.

arjun289 replied:
@Jeshan7 I understand that but, here the issues is rationg_option_vote and that can be inserted right?

Jeshan7 replied:
@arjun289 Ultimately rating_option_vote is nothing but  a map of rating_option as defined in the factory . So inserting rating_option_vote is same as inserting a rating_option.

arjun289 commented on apps/snitch_core/test/data/schema/review/review_test.exs:
```diff
@@ -0,0 +1,71 @@
+             } == errors_on(changeset)
+    end
+
+    test "fails for duplicate user_id", %{params: params} do
```
Test statement not correct, you have mentioned duplicate id but you are testing for non-existent one.

arjun289 commented on apps/snitch_core/test/data/schema/review/review_test.exs:
```diff
@@ -0,0 +1,71 @@
+  end
+
+  describe "update_changeset/2" do
+    test "returns a valid changset", %{review: review} do
```
typo `changeset`

Jeshan7 committed:
[#163622129]

Prior to this change,
  In review.ex (schema)
   - user_id was present in update_params .
  In product_review_test.exs (model)
   - Extra  IO.inspect statement was present .
  In review_test.exs (schema)
   - test statement was wrong.
   - spelling mistake in  the test statement.

This change
  In review.ex (schema)
   - removed user_id from update_params .
  In product_review_test.exs (model)
   - Removed IO.inspect statement .
  In review_test.exs (schema)
   - Corrected the test statement and some spellings .

Jeshan7 committed:
Merge branch 'develop' into review_tests

Jeshan7 committed:
[#163622129]

Prior to this change,
 - Needed to refactor review_test.exs(model)
 - In update/2 rating_option_vote of another review was being used.
This change
 - Refactored the review_test.exs(model)
 - In update/2 two rating_option_vote are now imported and used in the
   test cases.
